server/src/main/java/io/crate/action/sql/Sessions.java

```diff
@@ -153,6 +159,14 @@ public Session newSession(@Nullable String defaultSchema, Role authenticatedUser
         }
         sessionSettings.statementTimeout(defaultStatementTimeout);
         sessionSettings.memoryLimit(memoryLimit);
+
+        if (authenticatedUser.sessionSettings() != null) {
+            for (Map.Entry<String, Object> entry : authenticatedUser.sessionSettings().entrySet()) {
+                SessionSetting<?> setting = sessionSettingRegistry.settings().get(entry.getKey());
+                setting.apply(sessionSettings, entry.getValue());
```

Comment:
`settings().get(entry.getKey())` could in theory return `null` if we ever remove a session setting that's still persisted in the users default settings - depending on how we handle removal.

Maybe worth adding an assertion here that it's not null with a meaningful message.